src/form.js

```diff
@@ -897,14 +907,27 @@ define([
                 mug.options.afterInsert(this, mug);
             }
         },
-        _updateMugPath: function (mug, oldPath, newPath) {
-            var map = this.mugMap;
-            delete map[oldPath];
-            if (_.isUndefined(newPath)) {
+        _updateMugPath: function (mug, oldHashtag, newHashtag) {
+            var map = this.mugMap, newPath;
+            delete map[normalizeHashtag(oldHashtag)];
+            if (oldHashtag) {
+                xpath.removeHashtag(oldHashtag);
```

Comment:
To create a new xpath parser, the form constructor should do something like
```js
this.hashtagToXPath = {};
this.xpath = xpath.createParser(this.hashtagToXPath);
```
`xpath.createParser` should create the object that is currently exported by the `vellum/xpath` module.